README.md

```diff
@@ -41,6 +41,9 @@ This is the JavaScript version of [OpenTelemetry](https://opentelemetry.io/), a
 
 ## Quick Start
 
+**Much of OpenTelemetry JS documentation is written assuming the compiled application is run as CommonJS.**
+For more details on ECMAScript Modules vs CommonJS, refer to [esm-vs-cjs](https://github.com/open-telemetry/opentelemetry-js/blob/main/doc/esm-vs-cjs.md).
```

Comment:
(I am totally fine you not taking this suggestion/idea. I haven't thought it all through.)

We could consider naming the file "esm.md" or "esm-support.md". Then the focus of the doc is "How do I get OTel JS working with ESM code?"  It would still include all the content you've added for explaining the technical diffs between CJS and ESM.

Eventually, a hope of mine is that after (a) switching to `module.register(...)` usage (https://github.com/open-telemetry/opentelemetry-js/issues/4933) and (b) SDK 2.0 where we bump our min supported Node.js to effectively `^18.19.0 || ^20.6.0 || >=22` that we can have a **single** suggested bootstrap boilerplate that works for all users of CJS or ESM.  Then this doc/esm.md becomes a useful as a reference, but not a requirement for ESM users to learn.

Reply:
Yeah I agree with changing it to esm or esm-support; the esm-vs-cjs feels clunky.

Agreed I would love to have the recommendation be as simple as possible. In the meantime though I don't think we get that luxury.

Reply:
renamed to esm-support in a separate commit from the other changes, hopefully it doesn't make re-review too terrible.